app/workers/functions.php

```diff
@@ -115,65 +139,99 @@
                 ->addMetric(str_replace('{functionInternalId}', $function->getInternalId(), METRIC_FUNCTION_ID_EXECUTIONS), 1); // per function
         }
 
-        $execution->setAttribute('status', 'processing');
-        $execution = $dbForProject->updateDocument('executions', $executionId, $execution);
+        if ($execution->getAttribute('status') !== 'processing') {
+            $execution->setAttribute('status', 'processing');
+
+            if ($function->getAttribute('logging')) {
+                $execution = $dbForProject->updateDocument('executions', $executionId, $execution);
+            }
+        }
+
+        $durationStart = \microtime(true);
+
+        $vars = [];
 
-        $vars = array_reduce($function->getAttribute('vars', []), function (array $carry, Document $var) {
+        // Shared vars
+        $varsShared = $project->getAttribute('variables', []);
+        $vars = \array_merge($vars, \array_reduce($varsShared, function (array $carry, Document $var) {
+            $carry[$var->getAttribute('key')] = $var->getAttribute('value') ?? '';
+            return $carry;
+        }, []));
+
+        // Function vars
+        $vars = \array_merge($vars, array_reduce($function->getAttribute('vars', []), function (array $carry, Document $var) {
             $carry[$var->getAttribute('key')] = $var->getAttribute('value');
             return $carry;
-        }, []);
+        }, []));
 
-        /** Collect environment variables */
+        // Appwrite vars
         $vars = \array_merge($vars, [
             'APPWRITE_FUNCTION_ID' => $functionId,
             'APPWRITE_FUNCTION_NAME' => $function->getAttribute('name'),
             'APPWRITE_FUNCTION_DEPLOYMENT' => $deploymentId,
-            'APPWRITE_FUNCTION_TRIGGER' => $trigger,
             'APPWRITE_FUNCTION_PROJECT_ID' => $project->getId(),
             'APPWRITE_FUNCTION_RUNTIME_NAME' => $runtime['name'] ?? '',
             'APPWRITE_FUNCTION_RUNTIME_VERSION' => $runtime['version'] ?? '',
-            'APPWRITE_FUNCTION_EVENT' => $event ?? '',
-            'APPWRITE_FUNCTION_EVENT_DATA' => $eventData ?? '',
-            'APPWRITE_FUNCTION_DATA' => $data ?? '',
-            'APPWRITE_FUNCTION_USER_ID' => $user->getId() ?? '',
-            'APPWRITE_FUNCTION_JWT' => $jwt ?? '',
```

Comment:
lets make sure we mark these variables deprecated in https://appwrite.io/docs/functions#functionVariables